docs/docs/higher-order-components.md

```diff
@@ -0,0 +1,338 @@
+# Higher-order components
+
+A higher-order component (HOC) is an advanced technique in React for reusing component logic.
```

Comment:
I would stress that it is not a feature of React but a pattern right in the intro